cloud/scope/powervs_cluster.go

```diff
@@ -157,3 +261,1558 @@ func (s *PowerVSClusterScope) PatchObject() error {
 func (s *PowerVSClusterScope) Close() error {
 	return s.PatchObject()
 }
+
+// Name returns the CAPI cluster name.
+func (s *PowerVSClusterScope) Name() string {
+	return s.Cluster.Name
+}
+
+// Zone returns the cluster zone.
+func (s *PowerVSClusterScope) Zone() *string {
+	return s.IBMPowerVSCluster.Spec.Zone
+}
+
+// ResourceGroup returns the cluster resource group.
+func (s *PowerVSClusterScope) ResourceGroup() *infrav1beta2.IBMPowerVSResourceReference {
```

Comment:
again this * may create issues when something is wrong, wondering if we can have a simple null check?

Reply:
Added check where ever the value is being referred.